client/src/main/java/org/eclipse/hono/client/CommandResponse.java

```diff
@@ -34,8 +34,8 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(CommandResponse.class);
 
-    private static final Predicate<Integer> INVALID_STATUS_CODE = code ->
-        code == null || code < 200 || (code >= 300 && code < 400) || code >= 600;
+    private static final IntPredicate INVALID_STATUS_CODE = code ->
```

Comment:
this change seems to require *more* code being written ..
please revert this change.

Reply:
No, the idea is to not do this at all because it doesn't make things easier.